### ISSUE TITLE
Catalog inline anchors in callout lists?

### ISSUE THREAD

**Issue by owenh000** [asciidoctor/asciidoctor#2818]
Should inline anchors in callout list items be cataloged in the same manner as inline anchors in unordered and ordered lists?

See issue #2812 discussion, pull request [#2813 discussion](https://github.com/asciidoctor/asciidoctor/pull/2813/files/82503b4502ae353b513ac259c4cce8b878ac8753#r203256276), and pull request #2819.

**Comment by mojavelinux** [comment]
My understanding is that callout lists (colists) are there to support a verbatim block, such as a source listing. Is it really necessary to link directly to a callout? Is it even something we want to encourage?

I suppose one case would be that you want to link from the callout number in the source listing to the item in the callout list. But that's probably something you'd want to do with an extension. Otherwise, you'd end up with some pretty ugly and non-semantic markup to make it work.

So the open question is, what is the use case for linking directly to an item in a callout list. If the case is compelling, we can certainly entertain the idea of supporting it. But I don't want to add any overhead to the parser that is never going to be used.

**Comment by owenh000** [comment]
Here are my thoughts, with the following as an example.

```asciidoc
----
require 'sinatra' <1>
----
<1> [[import,Library import]]Library import

This is a cross-reference to <<import>>.
```

Current behavior (v1.5.7.1):

1. The inline anchor in the callout list item works (`<a id="import"></a>`).
2. The cross-reference to the anchor works (`<a href="#import">[import]</a>`)—yes, this is just HTML here.
3. However, the cross-reference is given a generic name (`[import]`) rather than the name specified (`Library import`).
4. Additionally, in verbose mode a warning is printed about an invalid reference, though it actually works (other than 3.).

Adding the cataloging fixes 3 and 4. Alternatively, if anchors are not supported in callout list items, I suppose they should not be parsed to begin with. In other words, the current behavior seems like inline anchors in callout lists are partially supported.

**Comment by mojavelinux** [comment]
Thanks @owenh000! I've included this in 2.0.
